msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/ManagedIdentityTests.java

```diff
@@ -118,7 +129,7 @@ void managedIdentityTest_SystemAssigned_SuccessfulResponse(ManagedIdentitySource
         IEnvironmentVariables environmentVariables = new EnvironmentVariablesHelper(source, endpoint);
         DefaultHttpClient httpClientMock = mock(DefaultHttpClient.class);
 
-        lenient().when(httpClientMock.send(eq(expectedRequest(source, resource)))).thenReturn(expectedResponse(200, getSuccessfulResponse(resource)));
+        when(httpClientMock.send(eq(expectedRequest(source, resource)))).thenReturn(expectedResponse(200, getSuccessfulResponse(resource)));
```

Comment:
@Avery-Dunn with removal of lenient all the tests will fail as the static caching is used. We should have a way to disable static caching for tests. 

Or add cache options like the dotnet have and make it package private for ManagedIdentityApplication